### PR TITLE
Represent similarity distance in dendrogram

### LINKED ISSUE #55: Represent the similarity distance between rows in dendrograms
Ask collaborators if we need to represent the similarity distance in the dendrogram views.

If we need to, then we have to modify the data structure to be able to convey the distance between pairs of rows.

### PR DESCRIPTION
**Pull request by sehilyi**
**Update**
- Visually encode similarity distance between nodes in dendrogram (Fixes #55, Toward #228)
   - Added a function `matrixToTreeWithDistance`, as well as a test for this
   - Added a minimal example (based on the previous `cistrome-track-2/rowInfo.json`) that stores `dist` values for hierarchical clustering results
   - Added x-axis to help interpreting the similarity distance

**Screenshot**
Screen Shot 2020-05-07 at 6 44 37 PM

### CODE REVIEW

File: src/demo/App.js [hunk 1]
```diff
@@ -166,6 +168,25 @@ const demos = {
             rowFilter: [ ]
         }
     },
+    "Minimumal Dataset": {
```
Comment (keller-mark) [inline src/demo/App.js:171]:
```suggestion
    "Minimal Dataset": {
```

File: src/demo/App.js [hunk 2]
```diff
@@ -166,6 +168,25 @@ const demos = {
...
+        viewConfig: hgDemoViewConfig2,
+        options: {
+            rowInfoAttributes: [
+                {field: "Hierarchical Clustering", type: "tree", position: "right"}
+            ],
+            rowFilter: [ ]
+        }
+    },
+    "Minimumal Dataset (w/ Dendrogram Simillarity Distance)": {
```
Comment (keller-mark) [inline src/demo/App.js:180]:
```suggestion
    "Minimal Dataset (w/ Dendrogram Similarity Distance)": {
```

File: src/TrackRowInfoVisDendrogram.js [hunk 3]
```diff
@@ -16,6 +16,7 @@ import TrackRowInfoControl from "./TrackRowInfoControl.js";
  * @prop {number} top The top position of this view.
  * @prop {number} width The width of this view.
  * @prop {number} height The height of this view.
+ * @prop {object[]} rowInfo Array of JSON objects, one object for each sample, without filtering/sorting based on selected rows.
```
Comment (keller-mark) [inline src/TrackRowInfoVisDendrogram.js:19]:
```suggestion
```

File: src/TrackRowInfoVisDendrogram.js [hunk 4]
```diff
@@ -28,6 +29,7 @@ import TrackRowInfoControl from "./TrackRowInfoControl.js";
 export default function TrackRowInfoVisDendrogram(props) {
     const {
         left, top, width, height,
+        rowInfo,
```
Comment (keller-mark) [inline src/TrackRowInfoVisDendrogram.js:32]:
```suggestion
```

**Comment by sehilyi** [comment]
Thank your for finding those issues!

**Pull request merged**